shuup/default_importer/importers/product.py

```diff
@@ -0,0 +1,142 @@
+# This file is part of Shuup.
+#
+# Copyright (c) 2012-2016, Shoop Ltd. All rights reserved.
+#
+# This source code is licensed under the AGPLv3 license found in the
+# LICENSE file in the root directory of this source tree.
+import six
+from django.db.models import ForeignKey
+from django.utils.translation import ugettext_lazy as _
+
+from shuup.core.models import (
+    Product, ProductType, ShopProduct, Supplier, TaxClass
+)
+from shuup.importer.exceptions import ImporterError
+from shuup.importer.importing import DataImporter, ImportMetaBase
+from shuup.simple_supplier.models import StockAdjustment
+from shuup.utils.properties import PriceProperty
+
+
+class ProductMetaBase(ImportMetaBase):
+    aliases = {
+        "type": ["product_type", "producttype"],
+        "tax_class": ["tax_class_name"],
+        "name": ["title"],
+        "keywords": ["tags"],
+        "qty": ["quantity", "stock_amount", "stock_qty", "stock_quantity"],
+        "suppliers": ["supplier"]
+    }
+
+    post_save_handlers = {
+        "handle_stocks": ["qty"],
+    }
+
+    def handle_stocks(self, fields, sess):
+        """
+        Handle stocks for product
+
+        If stock qty has been given, expect that a supplier with stock management must be available.
+        """
+        row = sess.row
+        supplier = row.get("supplier")
+        if not supplier:
+            raise ImporterError(_("Plase add supplier to row before importing stock quantities."))
+        # shuup only has 1 supplier support now so get first
+        obj = sess.importer.resolve_object(Supplier, supplier)
+        if not obj:
+            raise ImporterError(_("No supplier found, please check the supplier exists."))
+        if not obj.stock_managed:
+            raise ImporterError(_("This supplier doesn't handle stocks, please set Stock Managed on."))
+
+        for qty_field in self.aliases["qty"]:
+            val = row.get(qty_field, None)
+            if val is not None:
+                StockAdjustment(product=sess.instance, delta=val)
+                break
+
+    def presave_hook(self, sess):
+        # ensure tax_class id is there
+        product = sess.instance
+        if not product.tax_class_id:
+            product.tax_class_id = TaxClass.objects.first().pk
+
+        if not product.type_id:
+            product.type_id = ProductType.objects.first().pk
```

Comment:
I don't like this default. Could this default be set before the actual import if the tax_class information is not in the file.